src/aa_remove_data/pb_utils.py

```diff
@@ -0,0 +1,165 @@
+
+    def get_datastr(self, sample: type, year: int) -> str:
+        """Get a string contaiing information about a sample.
+        Args:
+            sample (type): A sample from a pb file.
+            year (int): The year the sample was collected.
+        Returns:
+            str: A string containing the sample information.
+        """
+        date = self.convert_to_datetime(year, sample.secondsintoyear)
+        return (
+            f"{date}    {sample.secondsintoyear:8d}    {sample.nano:9d}"
+            f"    {sample.val}\n"
+        )
+
+    def get_pv_type(self) -> str:
+        """Get the name of a pb file's pv type using information in its
+        header.
+
+        Returns:
+            str: Name of pv type, e.g VECTOR_DOUBLE.
+        """
+        type_descriptor = self.header.DESCRIPTOR.fields_by_name["type"]
+        enum_descriptor = type_descriptor.enum_type
+        return enum_descriptor.values_by_number[self.header.type].name
+
+    def get_proto_class(self) -> type:
+        """Get the EPICSEvent_pb2 class corresponding to the pv in a pb file.
+        Instances of this class can interpret pb messages of a matching type.
+
+        Returns:
+            type: EPICSEvent_pb2 protocol buffer class.
+        """
+        # Ensure self.pv_type is set first.
+        if not self.pv_type:
+            self.pv_type = self.get_pv_type()
+        pv_type_camel = self._get_proto_class_name()
+        proto_class = getattr(EPICSEvent_pb2, pv_type_camel)
+        return proto_class
+
+    def write_to_txt(self, filepath: PathLike):
+        """Write a text file from a PBUtils object.
+
+        Args:
+            filepath (PathLike): Filepath for file to be written.
+        """
+        pvname = self.header.pvname
+        year = self.header.year
+        pv_type = self.get_pv_type()
+        data_strs = [self.get_datastr(sample, year) for sample in self.samples]
+        with open(filepath, "w") as f:
+            # Write header
+            f.write(f"{pvname}, {pv_type}, {year}\n")
+            # Write column titles
+            f.write(f"DATE{' ' * 19}SECONDS{' ' * 5}NANO{' ' * 9}VAL\n")
+            # Write body
+            f.writelines(data_strs)
+
+    def read_pb(self, filepath: PathLike):
+        """Read a pb file that is structured in the Archiver Appliance format.
+        Gathers the header and samples from this file and assigns them to
+        self.header self.samples.
+
+        Args:
+            filepath (PathLike): Path to pb file.
+        """
+        with open(filepath, "rb") as f:
+            first_line = self._add_aa_escape_chars(f.readline().strip())
+            lines = f.readlines()
+        self.header.ParseFromString(first_line)
+        proto_class = self.get_proto_class()
+        self.samples = [proto_class() for n in range(len(lines))]
+        for i, sample in enumerate(self.samples):
+            line = self._add_aa_escape_chars(lines[i].strip())
```

Comment:
Its possible I misunderstand how this works. But I would have assumed you would be removing the aa_escape_chars to read and adding the aa_escape_chars back in to write?